src/cdomains/c2poDomain.ml

```diff
@@ -1281,13 +1281,16 @@ module D = struct
         match x,y with
         | None, None -> true
         | Some cc1, Some cc2 ->
-          (* add all terms to both elements *)
-          let terms = SSet.union (SSet.union cc1.set (BlDis.term_set cc1.bldis))
-              (SSet.union cc2.set (BlDis.term_set cc2.bldis)) in
-          let cc1, cc2 = Option.get (insert_set (Some cc1) terms), Option.get (insert_set (Some cc2) terms) in
-          equal_eq_classes cc1 cc2
-          && equal_diseqs cc1 cc2
-          && equal_bldis cc1 cc2
+          if cc1 == cc2 then
+            true
```

Comment:
You're right this is probably the better place for short-circuiting, as the wrappers outside may be different cells.

However, we can then remove the shortcut in 1277, pattern matching on an option is cheap!